test/test_filesystem_helper.cpp

```diff
@@ -187,6 +205,22 @@ TEST(TestFilesystemHelper, is_empty)
   EXPECT_TRUE(p.empty());
 }
 
+TEST(TestFilesystemHelper, exists)
+{
+  {
+    auto p = path("");
+    EXPECT_FALSE(p.exists());
+  }
+  {
+    auto p = path(".");
+    EXPECT_TRUE(p.exists());
+  }
```

Comment:
* what about `/`
* what about Windows specifics such as`C`, `C:`, `C:\`?

Reply:
Good point. Actually, we could have the same issue on *nix systems too: what is the parent path of "/"?

It seems clearly documented here: https://en.cppreference.com/w/cpp/filesystem/path/parent_path

The parent of "/" is "/". So, I think on Windows we should expect the parent of "C:\" is "C:\".

I've added tests to check for this 9a48203 (I think the logic is correct as-is).

Reply:
I don't know if we can assume that a drive with the letter "C" exists, but I think `\\` is also valid on Windows.

See 5c4cd01